lib/matplotlib/axes/_axes.py

```diff
@@ -3284,6 +3284,22 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         if len(x) != len(y):
             raise ValueError("'x' and 'y' must have the same size")
 
+        def check_if_negative(array):
+            try:
+                if np.any(array < 0):
+                    return True
+            except TypeError:   # Don't fail on 'datetime.*' types
```

Comment:
It's unclear to me why we wouldn't want to fail with datetime types.  After all you can't add datetimes so using them as x/yerr should fail, afaict (you can add a datetime to a timedelta, but timedeltas *can* be meaningfully compared to zero).

Reply:
Thanks @anntzer! Yeah, I added this line to not fail on `datetime.timedelta` https://github.com/matplotlib/matplotlib/blob/6d5659204c945b169596e12c5f47669471ab8edd/lib/matplotlib/tests/test_units.py#L174 
I've assumed that it's always positive. Let me see if there is a way to check that they are "meaningful compared to zero".

Reply:
Ah, I was thinking about np.timedelta, not datetime.timedelta, sorry for the careless reading (`np.timedelta(...) > 0` works).  Well, I guess my point remains: it would be nice to have the check also for datetime.timedelta inputs, but I don't know how easy that is.

Reply:
@anntzer I've added a check for timedelta types. However, I'm a bit uncomfortable with checking only `timedelta` type. `datetime.datetime` handles only positive values -> should not be an issue. What about `datetime.date`?

>you can add a datetime to a timedelta

Do you mean, that there is a way to convert anything to `timedelta` and check it? If yes, could you elaborate a bit more?

Reply:
I agree the special-casing is a bit annoying, I don't have any good solution to offer right now though.